### DISCUSSION TITLE
vipAddress in Conductor running in Kubernetes

### DISCUSSION THREAD

**Discussion by npk1994** [Netflix/conductor#2957] [Q&A]
Greetings everyone,

We are implementing the conductor server in Kuberenetes environment. We have running application pod masterdataservoce which exposes below endpoint.

http://mastedataservice:8080/mastedataservice/country/getAllByLocale/{code}

We are trying to use vipAddress for the http tasks. 

Task definition:

```
[{
  "name": "masterdata_country",
  "retryCount": 3,
  "retryLogic": "FIXED",
  "retryDelaySeconds": 10,
  "timeoutSeconds": 300,
  "responseTimeoutSeconds": 100,
  "timeoutPolicy": "TIME_OUT_WF",
  "ownerEmail": "<>"
}] 

```

We have configured a simple workflow to hit above mentioned endpoint

```
{
    "name": "masterdata_country_wf_using_vip",
    "description": "get the country details with code",
    "version": 2,
    "schemaVersion": 2,
	"ownerEmail": "<>",
    "tasks": [
        {
           "name":"masterdata_country",
           "taskReferenceName":"masterdata_country_2",
           "type":"HTTP",
           "inputParameters":{
              "http_request":{
                 "vipAddress":"masterdataservice",
                 "uri":"/masterdataservice/country/getAllByCodeAndLocale/${workflow.input.code}",
                 "method":"GET"
              }
           }
       }
    ]
}

```
Additionally we have enabled the spring Kuberenetes discovery at the conductor server for the service discovery.

But looks like the vipAddress is not resolved and we are getting below exceptions in Conductor server log:
1030790 [system-task-worker-1] ERROR com.netflix.conductor.tasks.http.HttpTask [] - Got unexpected http response - uri: 
"/masterdataservice/country/getAllByCodeAndLocale/IND, vipAddress: masterdataservice
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "/masterdataservice/country/getAllByCodeAndLocale/IND": null; nested exception is org.apache.http.client.ClientProtocolException"

We have tried below wf configuration which works fine.
```

{
    "name": "masterdata_country_wf",
    "description": "get the country details with code",
    "version": 2,
    "schemaVersion": 2,
	"ownerEmail": "<>",
    "tasks": [
        {
           "name":"masterdata_country",
           "taskReferenceName":"masterdata_country_2",
           "type":"HTTP",
           "inputParameters":{
              "http_request":{
                 "uri":"http://masterdataservice:8080/masterdataservice/country/getAllByCodeAndLocale/${workflow.input.code}",
                 "method":"GET"
              }
           }
       }
    ]
}

```
****Really appreciate any suggestion for the above problem.**
Thank you.**

**Comment by jxu-nflx** [comment]
Hello @npk1994 I took another look into the conductor oss code base, and it turns out that we don't support `vipAddress` for httptask. Even thought that field might be set, it was never used in httptask to make requests. Feel free to open a PR to add that function if needed.
